scripts/create-deploy-pr

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+ORIGIN=origin
+DEPLOY_BRANCH=stages/prod
+
+if [ "${CI:-}" == "" ]; then
+    echo "This script is meant to be run in a continuous integration environment."
+    exit 1
+fi
+
+SHA=$(curl --silent https://idp.staging.login.gov/api/deploy.json | jq -r .git_sha)
+echo "Preparing deploy PR for ${SHA}"
+
+LAST_RC=$(gh release list --exclude-drafts --exclude-pre-releases --limit 1 | grep -E --only-matching 'RC [0-9]+(\.[0-9]+)?' | sed 's/RC //')
+if [ "$LAST_RC" == "" ]; then
+    LAST_RC=0
+fi
+
+# TODO: Patch support
+LAST_RC_MAJOR=$(bc <<< "scale=0; ${LAST_RC} / 1")
+NEXT_RC=$((LAST_RC_MAJOR + 1))
+echo "Last RC was ${LAST_RC}. The next RC will be ${NEXT_RC}."
+
+RC_BRANCH=stages/rc-$(date +'%Y-%m-%d')
+
+if git rev-parse $RC_BRANCH > /dev/null 2>&1; then
+    echo "RC branch $RC_BRANCH already exists."
+    exit 1
+fi
+
+git checkout -b $RC_BRANCH
+git reset --hard $SHA
```

Comment:
I'm assuming there's a reason to do it this way rather than checking out main and then checking out that SHA and then doing the checkout -b. `git reset --hard` gets my attention because it can be destructive locally.

Reply:
I ended up just pushing the SHA to the rc branch directly rather than worrying about the local git repo at all.